File: packages/python/peru_spatial_utils/territorial/generate_recommendations.py
from typing import Iterable, List

from .recommendation_for import recommendation_for


def generate_recommendations(entities: Iterable[str]) -> List[str]:
    seen: set[str] = set()
    result: List[str] = []
    for entity in entities:
        if entity in seen:
            continue
        seen.add(entity)
        result.append(recommendation_for(entity))
    return result

File: packages/rust/src/territorial/generate_recommendations.rs
use super::recommendation_for::recommendation_for;

pub fn generate_recommendations(entities: &[&str]) -> Vec<String> {
    let mut seen: Vec<&str> = Vec::new();
    let mut out: Vec<String> = Vec::new();
    for entity in entities {
        if seen.contains(entity) {
            continue;
        }
        seen.push(entity);
        out.push(recommendation_for(entity));
    }
    out
}
